src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java

```diff
@@ -183,11 +183,12 @@ public TypoSpellChecker(Context context)
 
       // subscribe to spelling prefs changes (invalidateAll on changes)
       ValueChangeHandler<Boolean> prefChangedHandler = (event) -> context_.invalidateAllWords();
+      ValueChangeHandler<Boolean> realtimeChangedHandler = (event) -> loadDictionary();
```

Comment:
Is the dictionary load idempotent? (seems like if it isn't we'd want to make sure it isn't loaded and avoid loading when the pref changes to false)

Reply:
I should probably just rename the function. `loadDictionary` checks the values of all relevant preferences now before proceeding.